### ISSUE TITLE
Password obfuscation in multi-line statements

### ISSUE THREAD

**Issue by eperott** [Ericsson/ecaudit#180]
Obfuscation of passwords in multi-line statements was improved in #170. However, as reported by @smiklosovic in a comment, the solution was not bullet proof.

> For example, you can have a batch statement with multiple statements which are setting passwords. Even such a batch statement > is invalid, what we found is that it will be logged but the second password will not be obfuscated.

> Everything is in this ticket https://issues.apache.org/jira/browse/CASSANDRA-16669

Opening this new ticket to follow up on this

**Comment by 0rlych1kk4** [comment]
Hi,
I’d like to work on this issue.
From the description, it appears that password obfuscation may fail when multiple statements are present in batch or multi-line queries, resulting in subsequent credentials being logged in clear text.
My plan is to propose a fix to ensure that all password occurrences are consistently detected and obfuscated across multi-line and batch statements.

Before starting the implementation, could you please confirm:
1. Whether this issue is still relevant and open for contribution?
2. Which release branch the fix should target?

Thanks!
